tests/test_download.py

```diff
@@ -183,27 +184,28 @@ def test_download_all_one_fail(api, tmpdir, smallest_online_products):
         product_infos, triggered, failed_downloads = api.download_all(
             ids, str(tmpdir), max_attempts=1, checksum=True
         )
-        assert len(failed_downloads) == 1
-        assert len(triggered) == 0
-        assert len(product_infos) + len(failed_downloads) == len(ids)
+        exceptions = {k: v["exception"] for k, v in failed_downloads.items()}
+        assert sorted(failed_downloads) == [ids[0]], exceptions
+        assert type(list(exceptions.values())[0]) == InvalidChecksumError, exceptions
+        assert triggered == {}
+        assert sorted(list(product_infos) + list(failed_downloads)) == sorted(ids)
         assert id in failed_downloads
 
     tmpdir.remove()
 
 
+@pytest.mark.xfail(reason="The threading in this test seems to break VCR.py somehow")
```

Comment:
Hmm. We should fix this somehow, but that is a different issue to raise. Perhaps we can avoid Threading if `n_concurrent_dl` is 1.

Reply:
Let's revisit this if it pops up elsewhere later.